Extension/src/SSH/sshCommandRunner.ts

```diff
@@ -436,13 +436,14 @@ export function runInteractiveSshTerminalCommand(args: ITerminalCommandArgs): Pr
             hideFromUser: true
         };
 
-        let terminalDataHandlingQueue: Promise<void> = Promise.resolve();
-        terminalListener = systemInteractor.onDidWriteTerminalData(async e => {
+        terminalListener = systemInteractor.onDidStartTerminalShellExecution(async (e) => {
             if (e.terminal !== terminal) {
                 return;
             }
 
-            terminalDataHandlingQueue = terminalDataHandlingQueue.finally(() => void handleTerminalOutput(e));
+            for await (const data of e.execution.read()) {
+                void handleTerminalOutput(data);
```

Comment:
Should this be `void handleTerminalOutput(data).catch(logAndReturn.undefined);`?